### PR TITLE
Integrate support of NODE_EXTRA_CA_CERTS 

### PR DESCRIPTION
**Pull request by PeterSchafer**
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/CONTRIBUTING.md) rules

#### What does this PR do?
If a user specifies additional root CAs via the environment variable NODE_EXTRA_CA_CERTS, they get added to the internal Certificate Pool and forwarded to child processes.

Also Golang is updated to 1.19.4 for building and throughout the CI pipeline.

#### Where should the reviewer start?
* main.go 
* proxy.go

#### How should this be manually tested?
Run and look at "test/jest/acceptance/extra-certs.spec.ts"

#### Any background context you want to provide?
The Golang CLI uses NODE_EXTRA_CA_CERTS itself to set temporary certificates for child processes. If there are user defined certificates they need to be merged.

### CODE REVIEW

File: cliv2/cmd/cliv2/main.go [hunk 1]
```diff
@@ -89,10 +89,17 @@ func sendAnalytics(analytics analytics.Analytics, debugLogger *log.Logger) {
 
 	res, err := analytics.Send()
 	errorCodeReceived := res != nil && 200 <= res.StatusCode && res.StatusCode < 300
-	if err == nil && !errorCodeReceived {
+	if err == nil && errorCodeReceived {
```
Comment (Avishagp) [inline cliv2/cmd/cliv2/main.go:92]: This logic was changed but the logging didn't, is this right?

**Review by Avishagp — APPROVED** [review]

File: cliv2/cmd/cliv2/main.go (hunk 1 above)
Comment (PeterSchafer) [reply cliv2/cmd/cliv2/main.go:92]: Yes, I stumbled across this while debugging. The logic was incorrect and it resulted in an error message when actually being successful. I could actually have changed the variable name too. shall I do this?

Comment (Avishagp) [reply cliv2/cmd/cliv2/main.go:92]: Yes, I think we should! will make it more clear

Comment (PeterSchafer) [reply cliv2/cmd/cliv2/main.go]: done :)

**Pull request merged**